test/code_path/test.ml

```diff
@@ -47,5 +60,39 @@ let module M = struct
   in
   M.m
 [%%expect{|
-- : string = "Test"
+- : string =
+"(code_path(main_module_name Test)(submodule_path())(enclosing_module M)(value())(fully_qualified_path Test))"
+|}]
+
+module Outer = struct
+  module Inner = struct
+    let code_path = [%code_path]
+  end
+end
+let _ = Outer.Inner.code_path
+[%%expect{|
+module Outer : sig module Inner : sig val code_path : string end end
+- : string =
+"(code_path(main_module_name Test)(submodule_path(Outer Inner))(enclosing_module Inner)(value(code_path))(fully_qualified_path Test.Outer.Inner.code_path))"
+|}]
+
+module Functor() = struct
+  let code_path = ref ""
+  module _ = struct
+    let x =
+      let module First_class = struct
+        code_path := [%code_path]
+      end in
+      let module _ = First_class in
+      ()
+    ;;
+
+    ignore x
+  end
+end
+let _ = let module M = Functor() in !M.code_path
+[%%expect{|
+module Functor : functor () -> sig val code_path : string ref end
+- : string =
+"(code_path(main_module_name Test)(submodule_path(Functor _))(enclosing_module First_class)(value(x))(fully_qualified_path Test.Functor._.x))"
```

Comment:
This is interesting, right? So, if you store a code_path in a reference that is updated inside submodules/first class modules, the value of that reference will depend on the order in which the submodules/first class modules are compiled.

Reply:
Btw, on a different note: is it expected that `fully_qualified_path` is `Test.Functor._.x` here?

Reply:
> This is interesting, right? So, if you store a code_path in a reference that is updated inside submodules/first class modules, the value of that reference will depend on the order in which the submodules/first class modules are compiled.

I don't think that's right. The reference is only updated at runtime.

Reply:
> Btw, on a different note: is it expected that `fully_qualified_path` is `Test.Functor._.x` here?

Yes, that's what I expected it to be based on existing tests and documentation.